toolkit/ThreeWToolkit/data_visualization/plot_fft.py

```diff
@@ -0,0 +1,61 @@
+from typing import cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from .base_visualizer import BaseVisualizer
+
+
+class PlotFFT(BaseVisualizer):
+    def __init__(
+        self,
+        series: pd.Series,
+        title: str = "FFT Analysis",
+        sample_rate: float | None = None,
+    ) -> None:
+        self.series = series
+        self.title = title
+        self.sample_rate = sample_rate
+
+    def plot(self, ax: Axes | None = None) -> tuple[Figure, Axes]:
+        if self.series.empty:
+            raise ValueError("Input series is empty")
+
+        clean_series = self.series.dropna()
+        if clean_series.empty:
+            raise ValueError("Series contains only NaN values")
+
+        N = len(clean_series)
```

Comment:
More representative name for the variable `N`. Let's keep capitalized variables to constants.
```suggestion
        num_samples = len(clean_series)
```